src/Solvers/matrix_solver_utils.jl

```diff
@@ -154,12 +154,39 @@ function compute_matrix_for_linear_operation(::CPU, template_field, linear_opera
     return A
 end
 
+@kernel function compute_matrix_for_linear_operation_kernel!(Aeᵢⱼₖ, eᵢⱼₖ, linear_operation!, indices, values, args...)
+    i, j, k = @index(Global, NTuple)
+    Nx, Ny, Nz = size(eᵢⱼₖ)
+    N = Nx * Ny * Nz
+
+    if i <= Nx && j <= Ny && k <= Nz
+        parent(eᵢⱼₖ)  .= 0
+        parent(Aeᵢⱼₖ) .= 0
+
+        @inbounds eᵢⱼₖ[i, j, k] = 1
+
+        fill_halo_regions!(eᵢⱼₖ)
```

Comment:
unfortunately, it is not possible to call `fill_halo_regions!` inside a kernel. `fill_halo_regions!` is a function that _calls_ kernels and to call a kernel from a kernel you need dynamic parallelism (we do not make use of that in oceananigans). This has to do with the fact that a kernel is tied to a certain thread configuration, so calling a kernel from a kernel you might have to redefine the thread structure.